sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3820,6 +3820,17 @@ object SQLConf {
       .intConf
       .createWithDefault(ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH)
 
+  val ACCURATE_CASE_WHEN_NULLABILITY_CHECK =
+    buildConf("spark.sql.optimizer.accurateCaseWhenNullabilityCheck")
```

Comment:
a config is overkill IMO. Changing nullability from true to false is an optimization and it happens in query optimizer as well.

Reply:
Got it. I'll remove the flag.